pkg/aws/manager_ec2_vpc_peering_test.go

```diff
@@ -0,0 +1,226 @@
+package aws
+
+import (
+	"errors"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	"reflect"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/integr8ly/cluster-service/pkg/clusterservice"
+	"github.com/sirupsen/logrus"
+)
+
+func TestVpcPeeringManager_DeleteResourcesForCluster(t *testing.T) {
+	fakeLogger, err := fakeLogger(func(l *logrus.Entry) error {
+		return nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	type fields struct {
+		taggingClient func() *taggingClientMock
+		logger        *logrus.Entry
+		Ec2Api        ec2iface.EC2API
+	}
+	type args struct {
+		clusterId string
+		tags      map[string]string
+		dryRun    bool
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    []*clusterservice.ReportItem
+		wantErr string
+	}{
+		{
+			name: "successful run",
+			fields: fields{
+				Ec2Api: buildMockEc2Client(func(ec2Client *mockEc2Client) {
+					ec2Client.deleteVpcPeeringConnectionFn = func(input *ec2.DeleteVpcPeeringConnectionInput) (*ec2.DeleteVpcPeeringConnectionOutput, error) {
+						return &ec2.DeleteVpcPeeringConnectionOutput{}, nil
+					}
+				}),
+				taggingClient: func() *taggingClientMock {
+					client, err := fakeTaggingClient(func(c *taggingClientMock) error {
+						return nil
+					})
+					if err != nil {
+						t.Fatal(err)
+					}
+					return client
+				},
+				logger: fakeLogger,
+			},
+			args: args{
+				clusterId: fakeClusterId,
+				tags:      map[string]string{},
+				dryRun:    false,
+			},
+			want: []*clusterservice.ReportItem{
+				fakeReportItemDeleting(),
+			},
+		},
+		{
+			name: "do not fail when getting peering connection NotFound",
```

Comment:
Same q as above, how come we only test this for peering connections?

Reply:
^^